### ISSUE TITLE
feat: allow defining question templates with predefined tags, title, description

### ISSUE THREAD

**Issue by mugbug** [drodil/backstage-plugin-qeta#109]
## Proposal

Have a solution similar to what GitHub has with issue templates. The idea here is to facilitate standardization on questions created, to incentivize the use of specific tags for common category questions. For instance, if I'm going to ask a question related to a specific programming language or platform like Swift/iOS, I can choose a `iOS/Swift` template and it will come with a specific tag and a description requesting the person creating the question to give informations like the Swift version they are using. This could help on question filtering for example and also improving the details provided on questions.

## Screenshots

To use as reference, there is this [issue creation page](https://github.com/flutter/flutter/issues/new/choose) from an open source project:

<img width="1245" alt="image" src="https://github.com/drodil/backstage-plugin-qeta/assets/19307199/9cb3e3ee-3e77-4da5-90dd-856dbfa14ee6">

And when creating a new issue for, say, a [feature request](https://github.com/flutter/flutter/issues/new?assignees=&labels=&projects=&template=3_feature_request.yml), it already provides some requested information to be filled:

<img width="941" alt="image" src="https://github.com/drodil/backstage-plugin-qeta/assets/19307199/baa0268e-2526-4631-b6bb-6a8e50091f45">

Which is described via [this `yaml` file](https://github.com/flutter/flutter/blob/master/.github/ISSUE_TEMPLATE/3_feature_request.yml?plain=1).

**Comment by drodil** [comment]
Great idea! A couple of thoughts/questions:

- Should the templates be editable from the UI? If yes, then who can modify them? Should this be something that uses the backstage permission framework?
- If the templates cannot be edited from UI, I think a new config value is required to point to the folder where the templates are located
- A new parser is needed for the yaml file to render the form; should the [RJSF](https://rjsf-team.github.io/react-jsonschema-form/docs/) be used to render the form as it is already used by the software templates in Backstage?

Are there volunteers to work with this or should I throw this on my ever-growing personal backlog? 😆 All PRs are more than welcome!

**Comment by caioq** [comment]
I think it's a great idea to follow the same template approach as the backstage scaffolder.
Using templates as yaml files we could add spec definitions to standardize the questions.
In the `app-config.yaml` file, we could explicit the q&a templates path:
```yml
qeta:
  storage:
    type: database
  templates:
    - name: 'Flutter Questions'
       path: ./qeta-templates/flutter-question-template.yml
```

I'm afraid of using `react-jsonschema-form` because we might lose some features of the editor such as image upload and markdown support. If this is a problem, we could just have a markdown template, which will be pre-populated in the editor.

**Comment by rafaeljustinox** [comment]
It's a great ideia! Any news if this proposal is feasible? @drodil

**Comment by PankovSerge** [comment]
+1

**Comment by drodil** [comment]
@rafaeljustinox, yeah it's feasible (sorry for the delay, was on holiday), not sure though what is the best way to implement this; probably having the template definition in application configuration instead of path is a better way as it doesn't require changes to the packaging of the application. Contributions are welcome :)

**Comment by drodil** [comment]
Relates to #192 so taking this under work when starting with that!

**Comment by drodil** [comment]
This is now available in the release 3.4.0!

**Comment by mugbug** [comment]
That's awesome to hear, @drodil! Thanks a lot for it! 🎉
